extensions/SemanticScribunto/src/ScribuntoLuaLibrary.php

```diff
@@ -0,0 +1,365 @@
+	}
+
+	/**
+	 * This mirrors the functionality of the parser function #ask and makes it
+	 * available in lua.
+	 *
+	 * @since 1.0
+	 *
+	 * @param string|array $arguments parameters passed from lua, string or array depending on call
+	 *
+	 * @return array    array( null ) or array[]
+	 */
+	public function ask( $arguments = null ) {
+
+		$queryResult = $this->getLibraryFactory()->newQueryResultFrom(
+			$this->processLuaArguments( $arguments )
+		);
+
+		if ( !$this->isAQueryResult( $queryResult ) ) {
+			return [ $queryResult ];
+		}
+
+		$luaResultProcessor = $this->getLibraryFactory()->newLuaAskResultProcessor(
+			$queryResult
+		);
+
+		$result = $luaResultProcessor->getQueryResultAsTable();
+
+		return [ $this->convertArrayToLuaTable( $result ) ];
+	}
+
+	/**
+	 * Returns property type
+	 *
+	 * @since 1.0
+	 *
+	 * @param string $propertyName
+	 *
+	 * @return array
+	 */
+	public function getPropertyType( $propertyName = null ) {
+
+		$this->checkType( 'getPropertyType', 1, $propertyName, 'string' );
+		$propertyName = trim( $propertyName );
+
+		if ( $propertyName === '' ) {
+			return [ null ];
+		}
+
+		$property = DIProperty::newFromUserLabel( $propertyName );
+
+		if ( $property === null ) {
+			return [ null ];
+		}
+
+		return [ $property->findPropertyTypeID() ];
+	}
+
+	/**
+	 * Returns query results in for of the standard API return format
+	 *
+	 * @since 1.0
+	 *
+	 * @param string|array $arguments
+	 *
+	 * @return array
+	 */
+	public function getQueryResult( $arguments = null ) {
+
+		$queryResult = $this->getLibraryFactory()->newQueryResultFrom(
+			$this->processLuaArguments( $arguments )
+		);
+
+		if ( !$this->isAQueryResult( $queryResult ) ) {
+			return [ $queryResult ];
+		}
+
+		$result = $queryResult->toArray();
+
+		if( !empty( $result["results"] ) ) {
+			// as of now, "results" has page names as keys. lua is not very good, keeping non-number keys in order
+			// so replace string keys with the corresponding number, starting with 0.
+		    $result["results"] = array_combine( range( 0, count( $result["results"] ) - 1 ), array_values( $result["results"] ) );
+		}
+
+		return [ $this->convertArrayToLuaTable( $result ) ];
+	}
+
+	/**
+	 * This mirrors the functionality of the parser function #info and makes it
+	 * available to lua.
+	 *
+	 * @since 1.0
+	 *
+	 * @param string $text text to show inside the info popup
+	 * @param string $icon identifier for the icon to use
+	 *
+	 * @return string[]
+	 */
+	public function info( $text, $icon = 'info' ) {
+
+		// do some parameter processing
+		if ( ! trim( $text ) || ! is_string( $text ) ) {
+			// no info-text present, or wrong type. abort
+			return null;
+		}
+
+		// check if icon is set and valid
+		if ( !is_string( $icon ) || !in_array( $icon, [ 'note', 'warning' ] ) ) {
+			$icon = 'info';
+		}
+
+		// the actual info message is easy to create:
+		$result = smwfEncodeMessages(
+			[ $text ],
+			$icon,
+			' <!--br-->',
+			false // No escaping.
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Move this trailing comment on the previous empty line. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)